plugins/api-docs/src/components/AsyncApiDefinitionWidget/AsyncApiDefinition.tsx

```diff
@@ -146,7 +146,72 @@ type Props = {
   definition: string;
 };
 
+export const apis: AnyApiFactory[] = [
+  // ...
+
+  createApiFactory({
+    api: apiDocsConfigRef,
+    deps: {},
+    factory: () => {
+      // load the default widgets
+      const definitionWidgets = defaultDefinitionWidgets();
+      return {
+        getApiDefinitionWidget: (apiEntity: ApiEntity) => {
+          if (apiEntity.spec.type === 'asyncapi') {
+            return {
+              type: 'asyncapi',
+              title: 'AsyncAPI',
+              rawLanguage: 'yaml',
+              component: definition => (
+                <AsyncApiDefinitionWidget definition={definition} />
+              ),
+            } as ApiDefinitionWidget;
+          }
+
+          // fallback to the defaults
+          return definitionWidgets.find(d => d.type === apiEntity.spec.type);
+        },
+      };
+    },
+  }),
+];
+
+const fetchResolver = {
```

Comment:
Just to avoid things getting lost on Discord, I think we're good to go with this being passed as the resolver. I don't think we really need any additional configs or APIs?

Reply:
Implemented the PR just with the resolver https://github.com/backstage/backstage/pull/10975